instrumentation/kafka-clients-0.11/src/main/java/io/opentelemetry/auto/instrumentation/kafkaclients/KafkaDecorator.java

```diff
@@ -47,16 +48,21 @@ public String spanNameOnProduce(final ProducerRecord record) {
     return "destination";
   }
 
-  public void onConsume(final Span span, final ConsumerRecord record) {
+  public void onConsume(final Span span, final long startTimeMillis, final ConsumerRecord record) {
     span.setAttribute("partition", record.partition());
     span.setAttribute("offset", record.offset());
+    // don't record a duration if the message was sent from an old Kafka client
+    if (record.timestampType() != TimestampType.NO_TIMESTAMP_TYPE) {
+      final long produceTime = record.timestamp();
+      span.setAttribute("record.queue_time_ms", Math.max(0L, startTimeMillis - produceTime));
```

Comment:
I am not sure about this attribute:

- Is this naming convention consistent? Do we already have attributes with names in snake_case?
- Should we produce "derived" attributes? Instead of 2 separate attributes "startTime" and "produceTime".
- In any case would like to see at least a comment, explaining its meaning.

Reply:
> Is this naming convention consistent? Do we already have attributes with names in snake_case?

I see a couple existing with snake_case:
* `amqp.routing_key`
* `amqp.delivery_mode`
* `couchbase.operation_id`

I propose that we leave the name for now since we are already planning to review all non- semantic attributes for consistency / documentation.

> Should we produce "derived" attributes? Instead of 2 separate attributes "startTime" and "produceTime".

Since it's not a semantic attribute, backends won't understand it, so I think better to leave it in format that is most directly queryable/interesting to users (but this is me making stuff up, I haven't seen any related spec guidance around this).

> In any case would like to see at least a comment, explaining its meaning.

done